pkg/signal/signal.go

```diff
@@ -79,8 +79,10 @@ func (s Signal) Serialize() Serial {
 	return res
 }
 
-func (ser Serial) UpdateElem(idx int, newElem uint32) {
-	ser.Elems[idx] = elemType(newElem)
+func (ser Serial) AddElem(elem uint32, prio prioType) Serial {
```

Comment:
Do you really need the return value from here?

Reply:
Good point, thanks! I modified this (and the main canonicalizer/decanonicalizer funcs) to not need a return value